code/game/objects/items/weapons/shields.dm

```diff
@@ -43,7 +43,80 @@
 			to_chat(user, "<span class='notice'>It appears in decent condition, with some damage marks.</span>")
 		if(80 to 100)
 			to_chat(user, "<span class='notice'>It appears in perfect condition.</span>")
-	
+
+/obj/item/weapon/shield/riot/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(istype(I, /obj/item/stack/sheet/metal))
+		var/obj/item/stack/sheet/metal/metal_sheets = I
+		if(obj_integrity > max_integrity * 0.2)
+			return
+
+		if(metal_sheets.get_amount() < 1)
+			to_chat(user, "<span class='warning'>You need one metal sheet to repair the base of [src].</span>")
+			return
+
+		visible_message("<span class='notice'>[user] begins to repair the base of [src].</span>")
+
+		if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY) || obj_integrity >= max_integrity)
+			return
+
+		if(!metal_sheets.use(1))
+			return
+
+		repair_damage(max_integrity * 0.2)
+		visible_message("<span class='notice'>[user] repairs the base of [src].</span>")
+
+
+/obj/item/weapon/shield/riot/welder_act(mob/living/user, obj/item/I)
+	if(user.action_busy)
+		return FALSE
+
+	var/obj/item/tool/weldingtool/WT = I
+
+	if(!WT.isOn())
+		return FALSE
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return TRUE
```

Comment:
Every object has `current_acid` which will be set when acid is active on it.
Better than having to loop through all the items in a turf.

I also added a warning span around it to give it red text.
```suggestion
	if(current_acid)
		to_chat(user, "<span class='warning'>You can't get near that, it's melting!<span>")
		return TRUE
```